src/modules/workspaces/interceptors/workspaces-logs.interceptor.spec.ts

```diff
@@ -0,0 +1,699 @@
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { WorkspacesLogsInterceptor } from './workspaces-logs.interceptor';
+import { SequelizeWorkspaceRepository } from '../repositories/workspaces.repository';
+import {
+  WorkspaceLogType,
+  WorkspaceLogPlatform,
+  WorkspaceLogGlobalActionType,
+} from '../attributes/workspace-logs.attributes';
+import { CallHandler, ExecutionContext, Logger } from '@nestjs/common';
+import { lastValueFrom, of } from 'rxjs';
+import { WorkspaceItemType } from '../attributes/workspace-items-users.attributes';
+
+describe('WorkspacesLogsInterceptor', () => {
+  let interceptor: WorkspacesLogsInterceptor;
+  let workspaceRepository: DeepMocked<SequelizeWorkspaceRepository>;
+  const loggerDebugSpy = jest.spyOn(Logger, 'debug').mockImplementation();
+
+  beforeEach(async () => {
+    workspaceRepository = createMock<SequelizeWorkspaceRepository>();
+    interceptor = new WorkspacesLogsInterceptor(workspaceRepository);
+    jest.clearAllMocks();
+  });
+
+  describe('determinePlatform()', () => {
+    it('When client is drive-web, then it should return WEB platform', () => {
+      const platform = interceptor.determinePlatform('drive-web');
+      expect(platform).toBe(WorkspaceLogPlatform.Web);
+    });
+
+    it('When client is drive-mobile, then it should return MOBILE platform', () => {
+      const platform = interceptor.determinePlatform('drive-mobile');
+      expect(platform).toBe(WorkspaceLogPlatform.Mobile);
+    });
+
+    it('When client is drive-desktop, then it should return DESKTOP platform', () => {
+      const platform = interceptor.determinePlatform('drive-desktop');
+      expect(platform).toBe(WorkspaceLogPlatform.Desktop);
+    });
+
+    it('When client is unknown, then it should return UNSPECIFIED platform', () => {
```

Comment:
I assume this test should no longer be needed, as we are not allowing unknown platforms